packages/gatsby-plugin-sharp/src/worker.js

```diff
@@ -0,0 +1,48 @@
+const queue = require(`async/queue`)
+const { processFile } = require(`./process-file`)
+
+/** @typedef {import('./process-file').TransformArgs} TransformArgs */
+
+/**
+ * @typedef WorkerInput
+ * @property {string} inputPath the file path to transform
+ * @property {string} contentDigest
+ * @property {object} pluginOptions
+ * @property {{outputPath: string, transforms: TransformArgs[]}} transforms
+ */
+
+/**
+ * the queue concurrency is 1 as we only want to transform 1 file at a time.
+ * @param {(job: WorkerInput, callback: Function) => undefined} task
+ */
+const q = queue((job, callback) => {
+  Promise.all(
+    processFile(
+      job.inputPath,
+      job.contentDigest,
+      job.transforms.map(transform => {
+        return {
+          outputPath: transform.outputPath,
+          args: transform.transforms,
+        }
+      }),
+      job.pluginOptions
+    )
+  )
+    .then(() => callback())
+    .catch(err => callback(err))
```

Comment:
Let's add a test for failures to ensure that errors are propagated correctly from the worker to the calling file